clang/include/clang/Basic/DiagnosticCommonKinds.td

```diff
@@ -306,6 +306,10 @@ def err_opt_not_valid_without_opt : Error<
   "option '%0' cannot be specified without '%1'">;
 def err_opt_not_valid_on_target : Error<
   "option '%0' cannot be specified on this target">;
+def err_feature_not_valid_on_target : Error<
+  "feature '%0' cannot be specified on this target">;
+def err_mips_legacy_opt_not_valid_on_target : Error<
```

Comment:
Is "legacy" appropriate here? To NanoMips the flags are legacy, but from MIPS point of view they're still currently useful flags, and MIPS has more users than NanoMips. Would it be better to use the same message for 'not valid on target'?

Reply:
Eh, it makes totally sense to me. Thanks for pointing that out!